### DISCUSSION TITLE
Pass in sfdx-url as string in ``org login sfdx-url`` command

### DISCUSSION THREAD

**Discussion by k-capehart** [forcedotcom/cli#2586] [Ideas]
The ``sf org login sfdx-url`` command requires that the SFDX Authorization URL be read from a file using the ``-f`` flag. My proposal is that a new flag be added to pass in the URL as a string in the command itself.

Why:
This eliminates the need to create a transitory file in an automated CI/CD process. Instead, the URL can be stored exclusively (and securely) as a secret and passed in as a parameter to the command itself.

A PR for this has been created here: https://github.com/salesforcecli/plugin-auth/pull/872

**Comment by k-capehart** [comment]
Quick update on this:

- We cannot directly include the value in the command per security requirements
- An issue has been opened in oclif/core to allow flags to read from stdin, which will resolve this as we can enable this for the `sfdx-url` flag
- A PR has been created in oclif/core to get this implented, after which a new PR will be opened to enable this in salesforcecli/plugin-auth
https://github.com/oclif/core/issues/761

**Comment by mshanemc** [comment]
Delivered via the new `--sfdx-url-stdin` flag.  One bug fixed in this week's release.

https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#2286-feb-14-2024-stable-rc
